File: src/test/java/xyz/capybara/clamav/configuration/PlatformTest.java
package xyz.capybara.clamav.configuration;

import org.junit.Test;

import java.nio.file.Path;
import java.nio.file.Paths;

import static org.assertj.core.api.BDDAssertions.then;
import static org.junit.Assert.*;

public class PlatformTest {

    @Test
    public void should_convert_file_separator_to_UNIX() {
        // Given
        Platform platform = Platform.UNIX;
        Path originalPath = Paths.get("test\\path\\to\\file");
        // When
        String convertedPath = platform.toServerPath(originalPath);
        // Then
        then(convertedPath).isEqualTo("test/path/to/file");
    }

    @Test
    public void should_convert_file_separator_to_WINDOWS() {
        // Given
        Platform platform = Platform.WINDOWS;
        Path originalPath = Paths.get("test/path/to/file");
        // When
        String convertedPath = platform.toServerPath(originalPath);
        // Then
        then(convertedPath).isEqualTo("test\\path\\to\\file");
    }

    @Test
    public void should_keep_jvm_platform_file_separator() {
        // Given
        Platform platform = Platform.JVM_PLATFORM;
        Path originalPath = Paths.get("test/path/to/file");
        // When
        String convertedPath = platform.toServerPath(originalPath);
        // Then
        then(convertedPath).isEqualTo(originalPath.toString());
    }
}

File: src/test/kotlin/xyz/capybara/clamav/PlatformTest.kt
package xyz.capybara.clamav


import java.nio.file.Paths

import org.assertj.core.api.BDDAssertions.then
import org.junit.jupiter.api.Test

class PlatformTest {

    @Test
    fun `should convert file separator to UNIX`() {
        // Given
        val platform = Platform.UNIX
        val originalPath = Paths.get("test\\path\\to\\file")
        // When
        val convertedPath = platform.toServerPath(originalPath)
        // Then
        then(convertedPath).isEqualTo("test/path/to/file")
    }

    @Test
    fun `should convert file separator to WINDOWS`() {
        // Given
        val platform = Platform.WINDOWS
        val originalPath = Paths.get("test/path/to/file")
        // When
        val convertedPath = platform.toServerPath(originalPath)
        // Then
        then(convertedPath).isEqualTo("test\\path\\to\\file")
    }

    @Test
    fun `should keep JVM platform file separator`() {
        // Given
        val platform = Platform.JVM_PLATFORM
        val originalPath = Paths.get("test/path/to/file")
        // When
        val convertedPath = platform.toServerPath(originalPath)
        // Then
        then(convertedPath).isEqualTo(originalPath.toString())
    }
}
